tests/test_geometries.py

```diff
@@ -22,7 +22,7 @@ def assert_wkt_equivalence(geom: Union[Geometry, GeometryCollection]):
     """Assert WKT equivalence with Shapely."""
     # Remove any trailing `.0` to match Shapely format
     clean_wkt = re.sub(r"\.0(\D)", r"\1", geom.wkt)
-    assert shape(geom.dict()).wkt == clean_wkt
+    assert shape(geom).wkt == clean_wkt
```

Comment:
`shape` uses the `__geo_interface__` to construct so don't need `.dict()`. This also tests our implementation is usable by shapely.